src/client/datascience/interactive-window/nativeInteractiveWindow.ts

```diff
@@ -257,9 +259,10 @@ export class NativeInteractiveWindow implements IInteractiveWindowLoadable {
     }
 
     private registerKernel(notebookDocument: NotebookDocument, controller: VSCodeNotebookController) {
-        const kernel = this.kernelProvider.getOrCreate(notebookDocument.uri, {
+        const kernel = this.kernelProvider.getOrCreate(notebookDocument, {
```

Comment:
No ambiguity with the argument now, we pass the document, else if it was uri, one could accidentally provide the resource Uri (owning uri)